keps/sig-cli/20191008-kubectl-events.md

```diff
@@ -0,0 +1,160 @@
+---
+title: Kubectl events
+authors:
+  - "@hpandeycodeit"
+owning-sig: sig-cli
+participating-sigs:
+  - sig-cli
+reviewers:
+  - "@soltysh"
+  - "@pwittrock"
+approvers:
+  - "@soltysh"
+  - "@pwittrock"
+editor: TBD
+creation-date: 2019-10-08
+last-updated: 2019-10-08
+status: provisional
+see-also:
+  - n/a
+replaces:
+  - 
+superseded-by:
+  -
+---
+
+# kubectl events
+
+## Table of Contents
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+    - [Long standing issues](#long-standing-issues)
+  - [Goals](#goals)
+- [Proposal](#proposal)
+  - [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+- [Graduation Criteria](#graduation-criteria)
+<!-- /toc -->
+
+## Summary
+
+Presently, `kubectl get events` has some limitations. It cannot be extended to met the increasing user needs to 
+support more functionality without impacting the `kubectl get`. 
+
+For eg: Any modification to `--watch` functionality for `events` will also change the `--watch` for `kubectl get` since the `events` is dependent of `kubectl get`
+
+Some of the requested features for events include: 
+
+1. Extended behaviour for `--watch` 
+2. Default sorting of `events`
+
+This KEP proposes to add a new command `events` for `kubectl` and  be independent of `kubectl get`. This can be 
+extended to address the user requirements that cannot be achieved if the command is dependent of `get`.
+
+## Motivation
+
+A separate sub-command for `events` under `kubectl` which can help with long standing issues: 
+Some of these issues that be addressed with the above change are: 
+
+- User would like to see a stream of "create" or update events, filtering out deletes or any other event type. 
+- User would like to know when an object is deleted while `--watching` it. 
+- User would like to see all changes to a single object until it is deleted. 
+- User would like to watch an object until it exists. 
+- User would like to see the results of `events` in default sorting order. 
+- User would like to see a timeline of `events`
+
+Examples:
+
+- Proposed Kubectl events command
+  - `kubectl events`: Default sorted by .lastTimestamp
+     ```
+       LAST SEEN   TYPE      REASON    OBJECT        MESSAGE
+       57m         Normal    Pulling   pod/bad-pod   Pulling image "knginx"
+       7m31s       Warning   Failed    pod/bad-pod   Error: ImagePullBackOff
+       2m32s       Normal    BackOff   pod/bad-pod   Back-off pulling image "knginx" 
+      ```
+  - Following are the proposed options for the command: 
+
+  - `kuebctl events --watch-event=[]` flag that allows users to subscribe to particular events, 
```

Comment:
`--watch` and `--event` should be separate flags. The former is to stream events, similarly how it's done in `kubectl get`, the latter should allow filtering for specific event types.